edge/pkg/edged/edged_pods.go

```diff
@@ -737,7 +738,13 @@ func (e *edged) podFieldSelectorRuntimeValue(fs *v1.ObjectFieldSelector, pod *v1
 	case "spec.serviceAccountName":
 		return pod.Spec.ServiceAccountName, nil
 	case "status.hostIP":
-		hostIP, err := pkgutil.GetLocalIP(e.nodeName)
+		var hostIP string
+		var err error
+		if e.customInterfaceEnabled {
+			hostIP, err = e.getHostIPByInterface()
+		} else {
+			hostIP, err = pkgutil.GetLocalIP(e.nodeName)
+		}
```

Comment:
Can you try to use `netutil.ChooseBindAddressForInterface` instead?

Reply:
alright, will use this one instead.